core/src/main/java/org/elasticsearch/index/query/HasParentQueryParser.java

```diff
@@ -30,9 +30,10 @@
 public class HasParentQueryParser implements QueryParser<HasParentQueryBuilder>  {
 
     private static final HasParentQueryBuilder PROTOTYPE = new HasParentQueryBuilder("", EmptyQueryBuilder.PROTOTYPE);
-    private static final ParseField QUERY_FIELD = new ParseField("query", "filter");
-    private static final ParseField SCORE_FIELD = new ParseField("score_mode").withAllDeprecated("score");
-    private static final ParseField TYPE_FIELD = new ParseField("parent_type", "type");
+    public static final ParseField QUERY_FIELD = new ParseField("query", "filter");
+    public static final ParseField SCORE_MODE_FIELD = new ParseField("score_mode").withAllDeprecated("score");
```

Comment:
Looking further here after comparing to the docs:

The lower query here https://www.elastic.co/guide/en/elasticsearch/reference/master/query-dsl-has-parent-query.html parses well with the original code here https://github.com/elastic/elasticsearch/blob/148265bd164cd5a614cd020fb480d5974f523d81/core/src/main/java/org/elasticsearch/index/query/HasParentQueryParser.java#L34 but not when switching to SCORE_FIELD = new ParseField("score", "score_mode"). So apparently there's something I don't quite understand yet with how ParseField works.